js/sia.js

```diff
@@ -172,14 +172,22 @@ function SiadWrapper () {
       processOptions.stdio = [ 'ignore', out, err ]
     }
 
+    let daemonProcess
     // Spawn siad
-    const Process = require('child_process').spawn
-    var daemonProcess = new Process(settings.path, [
-      '--api-addr=' + settings.address,
-      '--rpc-addr=' + settings.rpcAddress,
-      '--host-addr=' + settings.hostAddress,
-      '--sia-directory=' + settings.datadir
-    ], processOptions)
+    try {
+      const Process = require('child_process').spawn
+      daemonProcess = new Process(settings.path, [
+        '--api-addr=' + settings.address,
+        '--rpc-addr=' + settings.rpcAddress,
+        '--host-addr=' + settings.hostAddress,
+        '--sia-directory=' + settings.datadir
+      ], processOptions)
+    } catch (e) {
+      if (callback !== null) {
+        callback(e)
```

Comment:
I think it is.  It is a usual javascript idiom to have completion callbacks with an `err` parameter, which is undefined if there is no error.

We are calling the callback with other errors, so I updated the docstring.